test/e2e/volume-provisioner/framework/framework.go

```diff
@@ -315,3 +302,373 @@ func (f *Framework) newConfigurationProvider(cfg *core.Config) (common.Configura
 	}
 	return conf, nil
 }
+
+// NewClientSetFromFlags builds a kubernetes client from flags.
+func NewClientSetFromFlags() (clientset.Interface, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", TestContext.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+	cs, err := clientset.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return cs, nil
+}
+
+// InstallVolumeProvisioner installs both block and fss volume provisioners and
+// waits for them to be ready.
+func InstallVolumeProvisioner(client clientset.Interface) error {
+	blockDeployment := &appsv1.Deployment{
```

Comment:
Are we switching to having the config for the components in code and not in yaml format? If so, we should make this clear to the team and stick to one approach to avoid having specs in two separate formats. Any particular reason why you chose in-line spec vs yaml? I would have personally opted for yaml since this is pretty static as far as I can tell. Same comment applies for the rest of the components.

Reply:
Its mainly a time / complexity tradeoff vs duplication.

I considered reading in the templated yaml from `dist/`, however, we include multiple resources in some files which would have made for relatively complex code.

This change means we'll have to reflect breaking changes to the yaml files in the test set-up, however, we make such changes relatively infrequently so it shouldn't be too burdensome.

Happy to explore reading in the yaml from `dist/` in the future though as it would be a better test but for now I'd like to get this in.